### PR TITLE
Dance Machine ~Fixes~

### PR DESCRIPTION
**Pull request by Robustin**
1) Dancing should no longer have a chance to leave people mobile on the floor 

2) Significant improvements to music processing should stop music from cutting out and improve performance (also stunned/dead people can no longer start to dance).

3)  More mob types can hear the music

4) Improvements were made to a couple dances so that they perform as intended

5) No more buzzing every time someone fails to access the machine

6) Disco lights now have a more immersive description

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Fixes https://github.com/tgstation/tgstation/issues/26390
Fixes https://github.com/tgstation/tgstation/issues/26389

**Comment by optimumtact** [comment]
150 plus hours of testing

**Comment by Robustin** [comment]
Only one of those is a bug and it only came about because Joan insisted that using initial pixel offsets (which is how freaking REST works) wasn't safe and I had to use inscrutable matrices, which turned out to be even more unsafe. 

Also how many contributors actually come back and proactively improve their shit. I'd reckon not many.

**Comment by ExcessiveUseOfCobblestone** [comment]
let's not turn a good fix into Ep. 420 of the never ending drama "the young and the coderbus"

**Comment by PKPenguin321** [comment]
Put my last comment into your OP @Robustin so the issues automatically close when this is merged

**Comment by Robustin** [comment]
Last time ill say this but its pretty obscure, prev_lying = 0 is what prevent matrix resets from giving you a bunch of floor monsters (there might still be issues but I wasn't able to create them with my patented + dance machine + 50 human flashbang test). There are still other things in the game that use matrix resets (wrestling belt) and this won't be the last time this issue shows up.

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -420,59 +424,13 @@
...
-/mob/living/carbon/proc/danceyhelper()
-	if (src)
-		animate(src, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
-	sleep (70)
-	if (src)
-		animate(src, transform = null, time = 1, loop = 0)
 
+/mob/living/proc/lying_fix()
+	animate(src, transform = null, time = 1, loop = 0)
+	lying_prev = 0
```
Comment (lzimann) [inline code/game/machinery/dance_machine.dm]: `FALSE`

Comment (Robustin) [inline code/game/machinery/dance_machine.dm]: Incorrect, lying_prev represents the number of degrees that the sprite is turned, between 0 and 360.

Comment (lzimann) [inline code/game/machinery/dance_machine.dm]: I love variables with a good explanatory name

**Comment by PKPenguin321** [comment]
You didn't put the Fixes lines into your op dork

**Pull request merged**